File: weather-demo-app-android/app/src/main/java/com/demo/weather/domain/model/RepositoryError.kt
package com.demo.weather.domain.model

sealed class RepositoryError : Exception() {
    class EmptyCity : RepositoryError()
    class Network : RepositoryError()
    class Unknown : RepositoryError()
}

fun RepositoryError.toMessage(): String {
    return when (this) {
        is RepositoryError.EmptyCity -> "Please enter a city"
        is RepositoryError.Network -> "Network error. Please try again."
        is RepositoryError.Unknown -> "Something went wrong"
    }
}

File: weather-demo-app-apple/Weather/Domain/Model/RepositoryError.swift
//
//  RepositoryError.swift
//  Weather
//
//  Created by Onur on 4/11/26.
//

enum RepositoryError: Error, Equatable {
    case emptyCity
    case network
    case unknown
}

extension RepositoryError {
    var message: String {
        switch self {
        case .emptyCity:
            return "Please enter a city"
        case .network:
            return "Network error. Please try again."
        case .unknown:
            return "Something went wrong"
        }
    }
}
